### PR TITLE
Adding CIFAR-10 DQN experiment

### PR DESCRIPTION
**Pull request by mordred-skywalker**
#### Reference issue
Contributes experiment related to #289

#### Type of change
Documentation - added experiment

#### What does this implement/fix?
This PR adds a Deep Q-learning experiment notebook on the task of image classification using CIFAR-10 dataset.

### CODE REVIEW

**Review by PSSF23 — CHANGES_REQUESTED** [review]
You are mixing `keras` and `tensorflow.keras` here. The later one should be always used.

**Comment by mordred-skywalker** [comment]
> You are mixing `keras` and `tensorflow.keras` here. The later one should be always used. Also, please rename notebook to `cifar_q_learning`.

Fixed!

**Review by PSSF23 — COMMENTED** [review]
Not completely I think. You don't need to install keras  from GitHub anymore right?

**Comment by mordred-skywalker** [comment]
> Not completely I think. You don't need to install keras from GitHub anymore right?

I don't think I have installed keras package in the notebook. Are you referring to the keras-rl2 package? That is an independent package for deep reinforcement learning algorithms. I am attaching the link here. ([https://github.com/taylormcnally/keras-rl2](url) ) 

You can also refer to Sarah's PR since we are using the same imported packages. [https://github.com/neurodata/ProgLearn/blob/staging/docs/experiments/Deep%20Q-learning%20in%20XOR-XNOR.ipynb](url)

**Comment by PSSF23** [comment]
@mordred-skywalker cool! Thanks for pointing that out. They use `tensorflow.keras` as well so it's all good.

**Comment by mordred-skywalker** [comment]
> @mordred-skywalker cool! Thanks for pointing that out. They use `tensorflow.keras` as well so it's all good. I will take another look over the notebook.

**Comment by mordred-skywalker** [comment]
> > @mordred-skywalker cool! Thanks for pointing that out. They use `tensorflow.keras` as well so it's all good. I will take another look over the notebook.

Sure! please let me know if other changes are required.

**Review by PSSF23 — COMMENTED** [review]
A lot of imported packages are not used. You can remove them and make the import code block shorter. Other than that LGTM.

**Comment by PSSF23** [comment]
Hmmm, it seems that you also need to work on GitHub skills as well, so that you don't delete & re-upload files each time, resulting in unnecessary commits. Try watching some video tutorials on how to commit & push your changes with GitHub Desktop or command line.

**Comment by mordred-skywalker** [comment]
> Hmmm, it seems that you also need to work on GitHub skills as well, so that you don't delete & re-upload files each time, resulting in unnecessary commits. Try watching some video tutorials on how to commit & push your changes with GitHub Desktop or command line.

Yes, I am relatively new to Github but I'll look into that :)

**Review by PSSF23 — CHANGES_REQUESTED** [review]
There are still unnecessary imports like `pandas`, `random`, & `shuffle`. Look through each line and just search in your file. If you did not use it, discard it.

**Comment by PSSF23** [comment]
> > Hmmm, it seems that you also need to work on GitHub skills as well, so that you don't delete & re-upload files each time, resulting in unnecessary commits. Try watching some video tutorials on how to commit & push your changes with GitHub Desktop or command line.
> 
> Yes, I am relatively new to Github but I'll look into that :)

You have been using it for a semester. Better make improvements before the next~

**Comment by mordred-skywalker** [comment]
> There are still unnecessary imports like `pandas`, `random`, & `shuffle`. Look through each line and just search in your file. If you did not use it, discard it.

Deleted them in the new commit.

**Review by PSSF23 — APPROVED** [review]

**Review by jdey4 — APPROVED** [review]
